src/main/src/main/java/org/geoserver/catalog/LayerGroupHelper.java

```diff
@@ -531,6 +543,29 @@ private static void expandStyleGroup(
         }
     }
 
+    /**
+     * Ensures that WMSLayer Info inside layergroup associates with correct remote style
+     *
+     * @return returns a compatible StyleInfo representing a remote style if required
+     */
+    private static StyleInfo processWMSLayer(WMSLayerInfo wmsLayerInfo, StyleInfo styleInfo) {
+
+        if (styleInfo == null) styleInfo = wmsLayerInfo.getDefaultStyle();
+        else if (!wmsLayerInfo.findRemoteStyleByName(styleInfo.getName()).isPresent()) {
+            if (LOGGER.isLoggable(Level.FINE)) {
```

Comment:
If the select style is not a valid style for this layer then an exception should be throw. Otherwise this may hide this fact and everything will look fine until it doesn't.